warehouse/packaging/search.py

```diff
@@ -33,7 +33,7 @@
 @doc_type
 class Project(Document):
 
-    name = Text()
+    name = Text(fields={'keyword': Keyword()})
```

Comment:
Here is where I add the inner field `name.keyword`, with the keyword type.
This is following @HonzaKral s suggestion.

Reply:
FYI - I saw no need to add a "deserialization" of `name.keyword` in the `from_db` class method. So I didn't do it :)

Reply:
That's not needed - the `name.keyword` does not acctually appear on the document, it is just an alternative index that can be used for searching and/or aggregations/sorting.